speechbrain/utils/checkpoints.py

```diff
@@ -75,6 +77,41 @@
 CKPT_PREFIX = "CKPT"
 METAFNAME = f"{CKPT_PREFIX}.yaml"  # Important that this is not .ckpt
 PARAMFILE_EXT = ".ckpt"  # ...because these files will be
+# some keys have been renamed in the new version of the code
+KEYS_MAPPING: Dict[str, str] = {
+    "mutihead_attn": "multihead_attn",  # see PR #2489
+}
+
+
+def map_old_state_dict_weights(
+    state_dict: Dict[str, torch.Tensor], mapping: Dict[str, str]
+) -> Dict[str, torch.Tensor]:
+    """
+    Maps the keys in the old state dictionary according to the provided mapping,
+    under the given prefix.
+
+    Parameters
+    ----------
+    state_dict : dict
+        The old state dictionary to be mapped.
+    mapping : dict
+        A dictionary specifying the mapping between old and new keys.
+
+    Returns
+    -------
+    dict
+        The modified state dictionary with mapped keys.
+    """
+    for checkpoint_name, attribute_name in mapping.items():
+        for full_checkpoint_name in list(state_dict.keys()):
+            if checkpoint_name in full_checkpoint_name:
+                full_attribute_name = full_checkpoint_name.replace(
```

Comment:
Should we have a warning here?

Reply:
Yep. Either a warning or a log in debug mode.